src/game.cpp

```diff
@@ -1715,6 +1715,17 @@ void game::set_critter_died()
     critter_died = true;
 }
 
+static int maptile_field_intensity( maptile &mt, field_type_id fld )
+{
+    auto field_ptr = mt.find_field( fld );
+
+    return ( field_ptr == nullptr ? 0 : field_ptr->get_field_intensity() );
+}
+static bool maptile_trap_eq( maptile &mt, const trap_id &id )
+{
+    return mt.get_trap_t().loadid == id;
```

Comment:
```suggestion
    return mt.get_trap() == id;
```
Should be the same, shouldn't it? And it avoids looking up the `trap_t` object.

Reply:
Absolutely correct! This was born of just translating the functionality of `map::tr_at`'s `current_submap->get_trap( l ).obj();` instead of thinking about the actual conversions happening.